builtin/logical/pki/path_config_ca.go

```diff
@@ -48,24 +470,57 @@ func (b *backend) pathCAWrite(
 		parsedBundle.CertificateBytes = parsedBundle.IssuingCABytes
 	}
 
-	// TODO?: CRLs can only be generated with RSA keys right now, in the
-	// Go standard library. The plubming is here to support non-RSA keys
-	// if the library gets support
+	cb := &certutil.CertBundle{}
+	entry, err := req.Storage.Get("config/ca_bundle")
+	if err != nil {
+		return nil, err
+	}
+	if entry != nil {
+		err = entry.DecodeJSON(cb)
+		if err != nil {
+			return nil, err
+		}
+		// If we have a stored private key and did not get one, attempt to
+		// correlate the two -- this could be due to a CSR being signed
+		// for a generated CA cert and the resulting cert now being uploaded
+		if len(cb.PrivateKey) != 0 &&
```

Comment:
I've gotten rid of reflect and moved the logic into a separate function in `certutil`. This code will eventually move to `/intermediate`, however, we still need this exact same logic when someone is uploading the signed key for an intermediate certificate, because we have to make sure that what they are uploading actually matches what we generated. So it may be a scary function to you, but we can't not perform these checks.